src/LuaTransformer.ts

```diff
@@ -5113,4 +5117,46 @@ export class LuaTransformer {
             return visitResult;
         }
     }
+
+    public generateConstructorDecorationStatement(
+        declaration: ts.ClassLikeDeclaration
+    ): tstl.AssignmentStatement | undefined {
+        const className = declaration.name !== undefined
+            ? this.transformIdentifier(declaration.name)
+            : tstl.createAnonymousIdentifier();
+
+        const decorators = declaration.decorators;
+        if (!decorators) { return undefined; }
+
+        const decoratorExpressions: tstl.Expression[] = [];
+        for (const decorator of decorators) {
+            const expressionVisitResult = this.transformExpression(decorator.expression);
+            if (expressionVisitResult !== undefined) {
+                decoratorExpressions.push(expressionVisitResult);
+            }
+        }
+
+        const decoratorArguments: tstl.Expression[] = [];
+
+        const decoratorTable = tstl.createTableExpression(
+            decoratorExpressions.map(expression => tstl.createTableFieldExpression(expression))
+        );
+
+        decoratorArguments.push(decoratorTable);
+        decoratorArguments.push(className);
+
+        const LuaLibDecorateIdentifier = this.importAndCreateIdentifierForLuaLibFunction(LuaLibFeature.Decorate);
```

Comment:
This (and the lines below) can probably just be `this.transformLuaLibFunction(LuaLibFeature.Decorate, className, ...decoratorArguments);`